peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
-        """Handle an interpolation."""
-        # The first arg is the string-like value "Hello " with a space
-        salutation = args[0].strip()
-        # The second arg is an "interpolation"
-        interpolation = args[1]
-        # Interpolations are tuples, the first item is a lambda
-        getvalue = interpolation[0]
-        # It gets called in the scope where it was defined, so
-        # the interpolation returns "World"
-        result = getvalue()
-        recipient = result.upper()
-        return f"{salutation} {recipient}!"
+    attributes = {"src": "shrubbery.jpg", "alt": "looks nice"}
+    template = t"<img {attributes} />"
+    assert html(template) == '<img src="shrubbery.jpg" alt="looks nice" />'
 
-    name = "World"
-    greeting = greet"Hello {name}"
-    assert greeting == "Hello WORLD!"
+Neither of these examples is possible with today's f-strings. By providing a 
+mechanism to intercept and transform interpolated values, template strings
+enable a wide range of string processing use cases.
 
-The f-string interpolation of ``{name}`` leads to the new machinery in tag
-strings:
 
-- ``args[0]`` is still the string-like ``'Hello '``, this time with a trailing space
-- ``args[1]`` is an expression -- the ``{name}`` part
-- Tag strings represent this part as an *interpolation* object as discussed below
+Specification
+=============
+
+Template String Literals
+------------------------
 
-The ``*args`` list is a sequence of ``Decoded`` and ``Interpolation`` values. A "decoded" object
-is a string-like object with extra powers, as described below. An "interpolation" object is a
-tuple-like value representing how Python processed the interpolation into a form useful for your
-tag function. Both are fully described below in `Specification`_.
+This PEP introduces a new string prefix, ``t``, to define template string literals.
+These literals resolve to a new type, ``types.Template``.
 
-Here is a more generalized version using structural pattern matching and type hints:
+The following code creates a ``Template`` instance:
 
 .. code-block:: python
 
-    from typing import Decoded, Interpolation  # Get the new protocols
+    from types import Template
+    template = t"This is a template string."
+    assert isinstance(template, Template)
 
-    def greet(*args: Decoded | Interpolation) -> str:
-        """Handle arbitrary args using structural pattern matching."""
-        result = []
-        for arg in args:
-            match arg:
-                case Decoded() as decoded:
-                    result.append(decoded)
-                case Interpolation() as interpolation:
-                    value = interpolation.getvalue()
-                    result.append(value.upper())
+Template string literals support the full syntax of :pep:`701`. This includes
+the ability to nest template strings within interpolations, as well as the ability
+to use any quote mark in the string literal. Like other string prefixes, the template
```

Comment:
Given this is within "Specification", perhaps enumerate (`'`, `"`, `'''`, `"""`) instead of "any quote mark"?